lib/qesapdeployment.pm

```diff
@@ -2311,6 +2312,51 @@ sub qesap_az_create_sas_token {
     return script_output($cmd);
 }
 
+=head2 qesap_az_list_container_files
+
+Returns a list of the files that exist inside a given path in a given container
+in Azure storage.
+
+Generated command looks like this:
+
+az storage blob list 
+--account-name <account_name> 
+--container-name <container_name> 
+--sas-token "<my_token>" 
+--prefix <path_inside_container> 
+--query "[].{name:name}" --output tsv
+
+=over 4
+
+=item B<STORAGE> - Storage account name used fur the --account-name argument in az commands
+
+=item B<CONTAINER> - container name within the storage account
+
+=item B<TOKEN> - name of the SAS token to access the account (needs to have l permission)
+
+=item B<PREFIX> - the local path inside the container (to list file inside a folder named 'dir', this would be 'dir')
+
+=back
+=cut
+
+sub qesap_az_list_container_files {
+    my (%args) = @_;
+    foreach (qw(storage container token prefix)) { croak "Missing mandatory $_ argument" unless $args{$_}; }
+    my $cmd = join(' ',
+        'az storage blob list',
+        '--account-name', $args{storage},
+        '--container-name', $args{container},
+        '--sas-token', $args{token},
+        '--prefix', $args{prefix},
+        '--query "[].{name:name}" --output tsv');
+    my $ret = script_output($cmd);
+    if (defined $ret && $ret ne '') {
```

Comment:
I think you can shorten it to:
```if ($ret && $ret ne ' ')```
To cover a case where script_output returns space character. 
```
cat ./test.pl 
my @test_values = ('', ' ', undef);


foreach (@test_values) {
        print "\n Value '$_' is true\n" if $_;
        print "\n Value '$_' is false\n" if !$_;
}
 perl ./test.pl 
 Value '' is false
 Value ' ' is true
 Value '' is false
```